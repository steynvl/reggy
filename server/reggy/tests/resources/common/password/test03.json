{
  "type": "Password",
  "information": {
    "shouldStartWith": "Letter or digit",
    "shouldContain": [
      "Digit",
      "Special character",
      "Minus",
      "Whitespace",
      "Lowercase letter",
      "Underline",
      "Uppercase letter"
    ],
    "minimumLength": "27",
    "maximumLength": "54"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
