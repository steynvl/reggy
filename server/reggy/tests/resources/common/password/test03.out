Java
Pattern regex = Pattern.compile("[A-Za-z0-9](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-)(?=.*\\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}");
[A-Za-z0-9](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-)(?=.*\\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
Perl
my $regex = /[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}/;
[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
POSIX
[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
Python
regex = re.compile(r'[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}')
[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
JavaScript
const regex = /[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}/;
[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
PHP
$regex = '/[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}/';
[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
Golang
regex, _ := regexp.Compile("[A-Za-z0-9](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-)(?=.*\\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}")
[A-Za-z0-9](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-)(?=.*\\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
Rust
let re = Regex::new(r"[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}").unwrap();
[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
C#
var re = new Regex(@"[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}");
[A-Za-z0-9](?=.*\d)(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*-)(?=.*\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
Scala
val re = "[A-Za-z0-9](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-)(?=.*\\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}".r
[A-Za-z0-9](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-)(?=.*\\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}
Kotlin
val regex = Regex("[A-Za-z0-9](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-)(?=.*\\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}")
[A-Za-z0-9](?=.*\\d)(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*-)(?=.*\\s)(?=.*[a-z])(?=.*_)(?=.*[A-Z]).{26,53}